### PR TITLE
IPC database service integration

### PR DESCRIPTION
**Pull request by NikVolf**
~~(just in case - **don't merge yet**, there are some issues with proper termination of executables)~~ - resolved

it leaves journaldb working as it is now
only blocks & extras db move to the ipc service
once some time passes, journaldb will move also and `BlockChain` and `Client` will be made database-agnostic

### CODE REVIEW

**Comment by debris** [comment]
Generally looks good, but I have one concern. imo it should be possible to prepare db `WriteBatch` entirely on client side.

**Comment by NikVolf** [comment]
@debris  resolved! (added `DBClientTransaction` and refactored where it can be critical to use it)

**Comment by gavofyork** [comment]
doesn't seem to play nice with `import`/`export`...

```
➜  parity git:(db-ipc-cont) ✗ RUST_BACKTRACE=1 cargo run --bin parity -- import ~/backup.rlp -d /tmp/x --format bin
     Running `target/debug/parity import /home/gav/backup.rlp -d /tmp/x --format bin`
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: BinaryConvertError', ../src/libcore/result.rs:746
stack backtrace:
   1:     0x5643729a7180 - sys::backtrace::tracing::imp::write::h3675b4f0ca767761Xcv
   2:     0x5643729ab55b - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.44519
   3:     0x5643729ab1c8 - panicking::default_handler::h18faf4fbd296d909lSz
   4:     0x56437299382c - sys_common::unwind::begin_unwind_inner::hfb5d07d6e405c6bbg1t
   5:     0x564372993cb8 - sys_common::unwind::begin_unwind_fmt::h8b491a76ae84af35m0t
   6:     0x5643729a65f1 - rust_begin_unwind
   7:     0x5643729dad4f - panicking::panic_fmt::h98b8cbb286f5298alcM
   8:     0x5643721c3f2c - result::unwrap_failed::h17096741098618386369
                        at ../src/libcore/macros.rs:29
   9:     0x5643721ec1e9 - result::Result<T, E>::unwrap::h12013326651589156376
                        at ../src/libcore/result.rs:687
  10:     0x5643721ea22b - database::DatabaseClient<S>.DatabaseService::open_default::h5412875023785767072
                        at target/debug/build/ethcore-db-f8ffc2f3778bad9e/out/lib.rs:2228
  11:     0x564371f5fb5b - blockchain::blockchain::BlockChain::new::h94706638722e6db0dyn
                        at ethcore/src/blockchain/blockchain.rs:267
  12:     0x564371f50103 - client::client::Client<V>::new_with_verifier::h13716044888553295759
                        at ethcore/src/client/client.rs:132
  13:     0x564371f4fd4d - client::client::Client<CanonVerifier>::new::h11618ad898be8032hGb
                        at ethcore/src/client/client.rs:106
  14:     0x564371f923c4 - service::ClientService::start::hcac0c745c1a0fed4Rsf
                        at ethcore/src/service.rs:64
  15:     0x5643717a2bb0 - execute_import::h7182564e2903c193iVc
                        at parity/main.rs:345
  16:     0x56437179bbae - execute::hb8a658171333ed07ECc
                        at parity/main.rs:121
  17:     0x56437179b863 - main::h745859a05066cde4sCc
                        at parity/main.rs:91
  18:     0x5643729aae24 - sys_common::unwind::try::try_fn::h14622312129452522850
  19:     0x5643729a657b - __rust_try
  20:     0x5643729aa8bb - rt::lang_start::h0ba42f7a8c46a626rKz
  21:     0x56437179cf29 - main
  22:     0x7fa4db7eea3f - __libc_start_main
  23:     0x56437179b6e8 - _start
  24:                0x0 - <unknown>
Process didn't exit successfully: `target/debug/parity import /home/gav/backup.rlp -d /tmp/x --format bin` (exit code: 101)
```

File: ipc/hypervisor/src/lib.rs [hunk 1]
```diff
@@ -121,6 +127,22 @@ impl Hypervisor {
 			worker.poll()
 		}
 	}
+
+	pub fn shutdown(&self, wait_time: Option<std::time::Duration>) {
+		if wait_time.is_some() { std::thread::sleep(wait_time.unwrap()) }
+
+		let mut childs = self.processes.write().unwrap();
+		for (ref mut binary, ref mut child) in childs.iter_mut() {
+			println!("Stopping {}", binary);
```
Comment (gavofyork) [inline ipc/hypervisor/src/lib.rs]: should be `info!`/`trace!` - don't want technical messages appearing to the user.

File: parity/db/main.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	worker.add_reqrep(addr).unwrap();
+	worker
+}
+
+fn main() {
+	let args: Args = Docopt::new(USAGE)
+							.and_then(|d| d.decode())
+							.unwrap_or_else(|e| e.exit());
+
+	println!("Database: {}", args.arg_path);
```
Comment (gavofyork) [inline parity/db/main.rs]: should be `info!`/`trace!` - don't want technical messages appearing to the user.

**Comment by gavofyork** [comment]
Performance:

with:
```
   #3869 e5ca…54e8    360 blk/s    0 tx/s   0 Mgas/s   12609+14102 Qed    0 bytes db    2 MiB chain   23 MiB queue
   #6261 3f5d…5ee0    419 blk/s    0 tx/s   0 Mgas/s    8286+21070 Qed    0 bytes db  516 KiB chain   35 MiB queue
   #8906 28c7…727c    505 blk/s    0 tx/s   0 Mgas/s    1997+27487 Qed    0 bytes db  573 KiB chain   26 MiB queue
  #12516 dfb8…c2e9    622 blk/s    0 tx/s   0 Mgas/s       1+29105 Qed    0 bytes db    1 MiB chain   23 MiB queue
  #15704 157f…150f    637 blk/s    0 tx/s   0 Mgas/s     110+29462 Qed    0 bytes db    1 MiB chain   23 MiB queue
  #18808 ffd9…9fee    620 blk/s    0 tx/s   0 Mgas/s      19+29702 Qed    0 bytes db    1 MiB chain   23 MiB queue
  #22717 5847…95c6    663 blk/s    0 tx/s   0 Mgas/s       1+29233 Qed    0 bytes db    2 MiB chain   23 MiB queue
  #28699 d812…d5bb    997 blk/s    0 tx/s   0 Mgas/s       1+28593 Qed    0 bytes db    3 MiB chain   24 MiB queue
  #35056 ae05…8069   1063 blk/s    0 tx/s   0 Mgas/s       1+28593 Qed    0 bytes db    5 MiB chain   24 MiB queue
  #39442 88f7…6e5e    853 blk/s    0 tx/s   0 Mgas/s       1+29105 Qed    0 bytes db    4 MiB chain   23 MiB queue
  #42491 8ed8…92ac    609 blk/s    0 tx/s   0 Mgas/s      28+29334 Qed    0 bytes db    3 MiB chain   23 MiB queue
  #45588 24b0…a798    619 blk/s    0 tx/s   0 Mgas/s       8+29690 Qed    0 bytes db    3 MiB chain   23 MiB queue
  #48924 fa27…2239    573 blk/s  242 tx/s   5 Mgas/s       1+29361 Qed    0 bytes db    3 MiB chain   24 MiB queue
  #51460 2265…9ab0    506 blk/s  182 tx/s   7 Mgas/s      92+29433 Qed    0 bytes db    3 MiB chain   23 MiB queue
  #53632 c2a0…561c    422 blk/s  313 tx/s  45 Mgas/s       1+29233 Qed    0 bytes db    3 MiB chain   23 MiB queue
  #56503 e0fe…9bdc    499 blk/s  236 tx/s  23 Mgas/s       1+29361 Qed    0 bytes db  293 KiB chain   23 MiB queue
  #60128 43a7…7bd1    725 blk/s  133 tx/s   3 Mgas/s      51+29633 Qed    0 bytes db  232 KiB chain   23 MiB queue
  #63987 e979…33bf    771 blk/s  189 tx/s   6 Mgas/s       7+29625 Qed    0 bytes db  197 KiB chain   23 MiB queue
  #67485 5219…7f6b    652 blk/s  302 tx/s  10 Mgas/s       1+28721 Qed    0 bytes db  646 KiB chain   23 MiB queue
  #69244 2e01…1528    333 blk/s  690 tx/s  24 Mgas/s       0+29617 Qed    0 bytes db  573 KiB chain   23 MiB queue
  #74239 7a30…ad50    887 blk/s  341 tx/s  10 Mgas/s       1+28721 Qed    0 bytes db    1 MiB chain   23 MiB queue
  #79330 1e1d…6e09    869 blk/s  403 tx/s  12 Mgas/s       1+29105 Qed    0 bytes db    2 MiB chain   23 MiB queue
  #84267 6d00…e93d    852 blk/s  465 tx/s  13 Mgas/s       1+28977 Qed    0 bytes db    3 MiB chain   24 MiB queue
  #89212 bfd9…9ca5    866 blk/s  444 tx/s  11 Mgas/s       1+28721 Qed    0 bytes db    3 MiB chain   23 MiB queue
  #91659 713f…4fa6    489 blk/s  299 tx/s   6 Mgas/s       1+29776 Qed    0 bytes db    2 MiB chain   23 MiB queue
  #96523 e4e0…bbb5    881 blk/s  438 tx/s  11 Mgas/s       1+28721 Qed    0 bytes db    3 MiB chain   23 MiB queue
 #100791 4b06…d0b4    853 blk/s  347 tx/s   7 Mgas/s     196+29645 Qed    0 bytes db    3 MiB chain   23 MiB queue
 #105033 c39c…e52b    719 blk/s  266 tx/s   7 Mgas/s       0+29231 Qed    0 bytes db    3 MiB chain   23 MiB queue
 #107835 5b38…10bb    473 blk/s  296 tx/s   7 Mgas/s       1+29361 Qed    0 bytes db  294 KiB chain   23 MiB queue
 #110208 4d30…2176    474 blk/s  148 tx/s   3 Mgas/s     107+29528 Qed    0 bytes db  195 KiB chain   23 MiB queue
 #112488 0cc2…0851    450 blk/s  327 tx/s  10 Mgas/s       0+29361 Qed    0 bytes db  231 KiB chain   23 MiB queue
 #114446 c9c1…192e    391 blk/s  267 tx/s   6 Mgas/s      59+29811 Qed    0 bytes db  195 KiB chain   23 MiB queue
 #116810 9a98…b492    444 blk/s  297 tx/s   8 Mgas/s       1+29617 Qed    0 bytes db  446 KiB chain   23 MiB queue
 #120763 b770…6747    776 blk/s  496 tx/s  12 Mgas/s       1+28721 Qed    0 bytes db  690 KiB chain   23 MiB queue
 #124626 5474…a13a    697 blk/s  596 tx/s  16 Mgas/s       1+29105 Qed    0 bytes db  892 KiB chain   23 MiB queue
 #127079 48ac…d33c    475 blk/s  312 tx/s   7 Mgas/s       1+29361 Qed    0 bytes db  776 KiB chain   23 MiB queue
 #129970 7395…38bf    571 blk/s  511 tx/s  12 Mgas/s       1+29233 Qed    0 bytes db  880 KiB chain   23 MiB queue
 #133216 ee82…ff51    581 blk/s  405 tx/s  12 Mgas/s       0+29489 Qed    0 bytes db  857 KiB chain   23 MiB queue
 #135121 ed0c…c01d    357 blk/s  314 tx/s   6 Mgas/s       1+29617 Qed    0 bytes db  723 KiB chain   24 MiB queue
 #138354 ed87…f37a    646 blk/s  549 tx/s  12 Mgas/s     249+29211 Qed    0 bytes db    1 MiB chain   24 MiB queue
 #142040 4ce0…451f    737 blk/s  551 tx/s  14 Mgas/s     108+29001 Qed    0 bytes db    2 MiB chain   24 MiB queue
 #145665 a5d6…a575    660 blk/s  602 tx/s  14 Mgas/s       1+29105 Qed    0 bytes db    1 MiB chain   23 MiB queue
 #147455 01ca…cc32    321 blk/s  370 tx/s   8 Mgas/s       1+29361 Qed    0 bytes db    1 MiB chain   23 MiB queue
 #150742 0ae4…c49c    657 blk/s  592 tx/s  13 Mgas/s      24+29238 Qed    0 bytes db    2 MiB chain   24 MiB queue
```


master:
```
  #16157 a5fd…82fa   1769 blk/s    0 tx/s   0 Mgas/s   25830+ 3185 Qed    0 bytes db  199 KiB chain   25 MiB queue
  #22976 08b3…4f47   1363 blk/s    0 tx/s   0 Mgas/s   23716+ 5223 Qed    0 bytes db    7 MiB chain   28 MiB queue
  #31862 acee…1f57   1746 blk/s    0 tx/s   0 Mgas/s   21567+ 7277 Qed    0 bytes db   10 MiB chain   28 MiB queue
  #40214 e36b…8ab9   1661 blk/s    0 tx/s   0 Mgas/s   19090+ 8859 Qed    0 bytes db    9 MiB chain   34 MiB queue
  #47254 636e…bbd8   1342 blk/s   88 tx/s   1 Mgas/s   18134+10838 Qed    0 bytes db    7 MiB chain   34 MiB queue
  #53685 5062…176a   1088 blk/s  589 tx/s  49 Mgas/s   13010+15962 Qed    0 bytes db    1 MiB chain   23 MiB queue
  #60253 436e…e34a   1313 blk/s  408 tx/s  31 Mgas/s   10772+18816 Qed    0 bytes db    1 MiB chain   35 MiB queue
  #66267 2b47…0278   1045 blk/s  401 tx/s  14 Mgas/s    6812+22288 Qed    0 bytes db    1 MiB chain   35 MiB queue
  #70449 da95…e8f6    734 blk/s  775 tx/s  25 Mgas/s      75+28512 Qed    0 bytes db    3 MiB chain   29 MiB queue
  #76471 223b…8c29   1202 blk/s  509 tx/s  17 Mgas/s     414+28717 Qed    0 bytes db    3 MiB chain   24 MiB queue
  #84263 40d0…a4e8   1551 blk/s  760 tx/s  22 Mgas/s       1+27953 Qed    0 bytes db    4 MiB chain   25 MiB queue
  #90533 eab7…f81c   1221 blk/s  731 tx/s  17 Mgas/s       1+29233 Qed    0 bytes db    1 MiB chain   24 MiB queue
  #94667 454d…7de3    813 blk/s  333 tx/s   8 Mgas/s       1+29361 Qed    0 bytes db    2 MiB chain   24 MiB queue
  #98970 6944…36d2    860 blk/s  339 tx/s   8 Mgas/s     171+29017 Qed    0 bytes db    2 MiB chain   24 MiB queue
 #102745 3f53…90ad    723 blk/s  397 tx/s   9 Mgas/s       1+29233 Qed    0 bytes db    2 MiB chain   24 MiB queue
 #105964 c28b…4198    610 blk/s  324 tx/s   8 Mgas/s       1+29361 Qed    0 bytes db    1 MiB chain   24 MiB queue
 #113377 2cc3…84a8   1482 blk/s  801 tx/s  22 Mgas/s     813+28444 Qed    0 bytes db    3 MiB chain   24 MiB queue
 #121255 27bb…b270   1342 blk/s  801 tx/s  20 Mgas/s       1+28721 Qed    0 bytes db    4 MiB chain   25 MiB queue
 #124452 9eed…8bda    639 blk/s  550 tx/s  16 Mgas/s     658+28993 Qed    0 bytes db    3 MiB chain   24 MiB queue
 #128501 86f1…b339    707 blk/s  470 tx/s  11 Mgas/s       1+28721 Qed    0 bytes db    3 MiB chain   24 MiB queue
 #131577 850d…cdcb    614 blk/s  475 tx/s  12 Mgas/s     425+29048 Qed    0 bytes db    3 MiB chain   24 MiB queue
 #134515 abb8…adac    536 blk/s  485 tx/s  13 Mgas/s       1+29233 Qed    0 bytes db    3 MiB chain   25 MiB queue
 #137271 a6a8…fb8c    490 blk/s  498 tx/s  11 Mgas/s       1+29617 Qed    0 bytes db    2 MiB chain   24 MiB queue
 #142735 c4a0…f326    962 blk/s  771 tx/s  19 Mgas/s       1+29361 Qed    0 bytes db    6 MiB chain   25 MiB queue
 #147058 32be…8c0c    778 blk/s  701 tx/s  16 Mgas/s       1+28849 Qed    0 bytes db  444 KiB chain   23 MiB queue
 #150664 a1c5…83d3    610 blk/s  584 tx/s  13 Mgas/s       1+29233 Qed    0 bytes db    1 MiB chain   24 MiB queue
```

**Comment by NikVolf** [comment]
@gavofyork i think introducing one obvious optimization will be the next step, it should not take long

**Comment by gavofyork** [comment]
```
   #5177 ff77…3602    514 blk/s    0 tx/s   0 Mgas/s   12133+17273 Qed    0 bytes db    3 MiB chain   35 MiB queue
   #8155 4ccf…9c34    546 blk/s    0 tx/s   0 Mgas/s    5190+23782 Qed    0 bytes db  408 KiB chain   29 MiB queue
  #12328 4fa9…a314    697 blk/s    0 tx/s   0 Mgas/s       1+29105 Qed    0 bytes db    1 MiB chain   23 MiB queue
  #17437 b272…5579   1003 blk/s    0 tx/s   0 Mgas/s       1+28465 Qed    0 bytes db    2 MiB chain   23 MiB queue
  #21818 c9d8…f2c3    876 blk/s    0 tx/s   0 Mgas/s     140+29540 Qed    0 bytes db  683 KiB chain   23 MiB queue
  #24252 a70b…44ec    486 blk/s    0 tx/s   0 Mgas/s     121+29691 Qed    0 bytes db  638 KiB chain   23 MiB queue
  #26736 c970…48a7    496 blk/s    0 tx/s   0 Mgas/s      93+29614 Qed    0 bytes db  561 KiB chain   23 MiB queue
  #29303 746c…2ea9    513 blk/s    0 tx/s   0 Mgas/s      74+29523 Qed    0 bytes db  519 KiB chain   23 MiB queue
  #31862 acee…1f57    511 blk/s    0 tx/s   0 Mgas/s     341+29511 Qed    0 bytes db  464 KiB chain   24 MiB queue
  #34343 c494…0839    496 blk/s    0 tx/s   0 Mgas/s      40+29722 Qed    0 bytes db  363 KiB chain   23 MiB queue
  #36889 18f6…531c    509 blk/s    0 tx/s   0 Mgas/s      67+29540 Qed    0 bytes db  442 KiB chain   23 MiB queue
  #39388 8c32…964c    494 blk/s    0 tx/s   0 Mgas/s       1+29233 Qed    0 bytes db  654 KiB chain   23 MiB queue
  #41988 877c…197e    501 blk/s    0 tx/s   0 Mgas/s       0+29233 Qed    0 bytes db  904 KiB chain   23 MiB queue
  #44491 f82c…ef52    494 blk/s    0 tx/s   0 Mgas/s       1+29233 Qed    0 bytes db  710 KiB chain   23 MiB queue
  #46909 ccb8…8c34    473 blk/s   60 tx/s   1 Mgas/s       1+29489 Qed    0 bytes db  710 KiB chain   23 MiB queue
  #48979 398a…97db    413 blk/s  224 tx/s   4 Mgas/s      42+29556 Qed    0 bytes db  560 KiB chain   23 MiB queue
  #51225 196d…5747    449 blk/s  155 tx/s   4 Mgas/s      31+29486 Qed    0 bytes db  594 KiB chain   23 MiB queue
  #53033 3b8e…47ba    329 blk/s  288 tx/s  42 Mgas/s       0+29617 Qed    0 bytes db  515 KiB chain   23 MiB queue
  #56724 1de7…20d9    635 blk/s  266 tx/s  25 Mgas/s       1+28721 Qed    0 bytes db    2 MiB chain   23 MiB queue
  #61671 91b5…ceb4    989 blk/s  213 tx/s   6 Mgas/s      10+29399 Qed    0 bytes db    2 MiB chain   24 MiB queue
  #66131 b1f9…46bc    771 blk/s  325 tx/s  11 Mgas/s       1+29105 Qed    0 bytes db    2 MiB chain   23 MiB queue
  #68640 64ef…2b7d    427 blk/s  408 tx/s  13 Mgas/s       0+29873 Qed    0 bytes db    1 MiB chain   23 MiB queue
  #71160 051a…d5a5    479 blk/s  453 tx/s  14 Mgas/s       1+29233 Qed    0 bytes db    2 MiB chain   23 MiB queue
  #73701 e893…26d6    508 blk/s  169 tx/s   6 Mgas/s      61+29741 Qed    0 bytes db    2 MiB chain   24 MiB queue
  #76985 0836…b3a1    560 blk/s  246 tx/s   8 Mgas/s       0+29233 Qed    0 bytes db    3 MiB chain   23 MiB queue
  #79377 d53d…aa64    478 blk/s  221 tx/s   5 Mgas/s      68+29632 Qed    0 bytes db    3 MiB chain   23 MiB queue
  #82460 dc00…a6cf    518 blk/s  315 tx/s  10 Mgas/s       1+29233 Qed    0 bytes db    3 MiB chain   23 MiB queue
  #85377 bd7b…4f59    487 blk/s  336 tx/s   7 Mgas/s       1+29233 Qed    0 bytes db  728 KiB chain   23 MiB queue
  #87981 30a1…60a8    520 blk/s  203 tx/s   5 Mgas/s      21+29491 Qed    0 bytes db  614 KiB chain   23 MiB queue
  #90955 e099…bde5    516 blk/s  276 tx/s   6 Mgas/s       1+29233 Qed    0 bytes db    1 MiB chain   24 MiB queue
  #93494 7afb…71e3    507 blk/s  147 tx/s   4 Mgas/s      27+29374 Qed    0 bytes db 1000 KiB chain   23 MiB queue
  #96415 f67c…f5e6    488 blk/s  314 tx/s   7 Mgas/s       1+29233 Qed    0 bytes db    1 MiB chain   23 MiB queue
  #99145 bfae…d484    529 blk/s  133 tx/s   3 Mgas/s       0+29105 Qed    0 bytes db    2 MiB chain   23 MiB queue
 #101367 f930…4d9b    444 blk/s  271 tx/s   6 Mgas/s      49+29416 Qed    0 bytes db    1 MiB chain   23 MiB queue
 #104800 2b1a…0c91    686 blk/s  242 tx/s   7 Mgas/s      60+29635 Qed    0 bytes db    2 MiB chain   23 MiB queue
 #110332 2c69…e3b4   1003 blk/s  495 tx/s  12 Mgas/s       1+28721 Qed    0 bytes db    3 MiB chain   24 MiB queue
 #113367 9a9c…d86a    527 blk/s  407 tx/s  12 Mgas/s       1+29233 Qed    0 bytes db    2 MiB chain   24 MiB queue
 #117467 71cd…a651    734 blk/s  445 tx/s  12 Mgas/s       1+29361 Qed    0 bytes db    3 MiB chain   24 MiB queue
 #121101 8b56…ba36    640 blk/s  390 tx/s   9 Mgas/s       0+28721 Qed    0 bytes db    3 MiB chain   23 MiB queue
 #125301 1770…876a    758 blk/s  608 tx/s  17 Mgas/s       1+28977 Qed    0 bytes db  261 KiB chain   23 MiB queue
 #128663 f35c…0b2e    613 blk/s  557 tx/s  13 Mgas/s       1+29745 Qed    0 bytes db  325 KiB chain   23 MiB queue
 #131577 850d…cdcb    511 blk/s  263 tx/s   7 Mgas/s       1+29361 Qed    0 bytes db  686 KiB chain   23 MiB queue
 #133689 782a…bf93    414 blk/s  360 tx/s  11 Mgas/s       1+29233 Qed    0 bytes db  964 KiB chain   23 MiB queue
 #135566 4943…0905    350 blk/s  349 tx/s   7 Mgas/s       1+29489 Qed    0 bytes db  935 KiB chain   23 MiB queue
 #137266 5f93…beb8    321 blk/s  319 tx/s   7 Mgas/s       1+29617 Qed    0 bytes db  562 KiB chain   23 MiB queue
 #139637 63a5…75c0    398 blk/s  261 tx/s   6 Mgas/s       1+29873 Qed    0 bytes db    1 MiB chain   23 MiB queue
 #141818 207a…8f4c    426 blk/s  369 tx/s   8 Mgas/s       0+29233 Qed    0 bytes db    2 MiB chain   23 MiB queue
 #143577 93d4…379a    348 blk/s  375 tx/s  10 Mgas/s       1+29233 Qed    0 bytes db    1 MiB chain   23 MiB queue
 #145826 0b57…3ba1    385 blk/s  292 tx/s   6 Mgas/s       1+29873 Qed    0 bytes db    1 MiB chain   24 MiB queue
 #148010 d1f7…88e9    417 blk/s  418 tx/s  10 Mgas/s       1+29489 Qed    0 bytes db    2 MiB chain   24 MiB queue
 #150702 7ddb…1551    533 blk/s  507 tx/s  11 Mgas/s       1+29233 Qed    0 bytes db    3 MiB chain   24 MiB queue
```

**Pull request closed without merging**